### PR TITLE
Code fix should prefer `import type`

### PR DESCRIPTION
**Pull request by andrewbranch**
Noticed this while reviewing @johnnyreilly’s [post](https://johnnyreilly.com/typescript-5-importsnotusedasvalues-error-eslint-consistent-type-imports) about `importsNotUsedAsValues`, `verbatimModuleSyntax`, and eslint.

`importsNotUsedAsValues` would produce errors any time the whole import declaration wasn’t marked as type-only but could be, so the error span was over the whole import, and the only fix that would satisfy it was to use a top-level `import type`. `verbatimModuleSyntax`, on the other hand, cares more about whether the individual import specifiers are safe to emit, so the error span from that option is placed on individual import specifiers, and using a `type` modifier on each specifier can satisfy the check. However, under `verbatimModuleSyntax`, it’s generally _preferable_ to use a top-level `import type` on the whole import declaration when possible, and the built-in codefix hadn’t been updated to consider that. Even though it’s no longer a TS error to have an import that will unnecessarily be emitted to JS, default behavior from auto-imports and codefixes should avoid that.

For individual fixes, both fix options are offered if `import type` is possible, with the `import type` being sorted first:

image

Using `import type` is not possible when other specifiers are already used as values:

image

For fix-all, `import type` is used whenever possible.

### CODE REVIEW

File: src/services/codefixes/convertToTypeOnlyImport.ts [hunk 1]
```diff
@@ -30,16 +37,42 @@ registerCodeFix({
...
+                changes,
+                declaration.kind === SyntaxKind.ImportSpecifier
+                    ? [Diagnostics.Use_type_0, declaration.propertyName?.text ?? declaration.name.text]
+                    : Diagnostics.Use_import_type,
+                fixId,
+                Diagnostics.Fix_all_with_type_only_imports);
+
+            if (some(importDeclarationChanges)) {
+                return [
+                    createCodeFixActionWithoutFixAll(fixId, importDeclarationChanges, Diagnostics.Use_import_type),
```
Comment (andrewbranch) [inline src/services/codefixes/convertToTypeOnlyImport.ts:54]: I’m not sure if the two fix options should have different `fixName`s

**Review by jakebailey — APPROVED** [review]
LGTM, though it seems like our `Convert ...` language is somewhat universal; is it worth preserving that wording?

**Comment by andrewbranch** [comment]
I think it’s usually better to quote exact syntax, and “type-only import” is ambiguous between type-only import declarations and type-only import specifiers. Not sure how to preserve the word “convert” and disambiguate, and I also don’t think I like the word to describe just adding a modifier.

File: src/services/codefixes/convertToTypeOnlyImport.ts (hunk 1 above)
Comment (sheetalkamat) [reply src/services/codefixes/convertToTypeOnlyImport.ts:54]: Change looks good to me but i don't know about this either.

Comment (andrewbranch) [reply src/services/codefixes/convertToTypeOnlyImport.ts:54]: Asked at the editor sync and this is only used for telemetry, and the telemetry stakeholders don’t care if they have the same name.

**Comment by andrewbranch** [comment]
I’m going to add a fix-all test before merging.

**Pull request merged**